### PR TITLE
Fix upgrade format error handling and validation

### PR DESCRIPTION
**Pull request by Werni**

### CODE REVIEW

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 85e83e80bdfb493c93f63d8bfcec5846deec433e into 78f97d20bdd3279abfbd7a7c32df05d3c52f3a92 )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 53e1609db3cbb419277c5e04efd99df396428c3d into 78f97d20bdd3279abfbd7a7c32df05d3c52f3a92 )

File: lib/upgrade_format_task.rb [hunk 1]
```diff
@@ -29,28 +29,28 @@ def upgrade(store, name, options = {})
...
 
-    migrations_done = 0
+    errors = []
+
     descriptions.each do |description|
       begin
         Machinery.logger.info "Upgrading description \"#{description}\""
-        migrated = Migration.migrate_description(store, description, force: options[:force])
-        migrations_done += 1 if migrated
+        Migration.migrate_description(store, description, force: options[:force])
```
Comment (cornelius) [inline lib/upgrade_format_task.rb]: If you don't check the return value anymore you miss the case where the description doesn't need to be upgraded. Then you get the message that the description is up to date as well as that the description is upgraded.

**Comment by cornelius** [comment]
This patch doesn't do what it should do. For a case when you have a description in format 1 which validates in format 2, but not in format 1, it upgrades without `--force` option.

**Comment by cornelius** [comment]
Maybe it would be good to split up the pull request and merge the refactorings separately.

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 53e1609db3cbb419277c5e04efd99df396428c3d into beb797124d1f510110c3765a18b40405c0071cef )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 53e1609db3cbb419277c5e04efd99df396428c3d into bfffd41f64708d7ffa147994035dd519ea5009aa )

**Comment by suse-tests-pass** [comment]
Well Done! Your tests are still passing.
( Success: Merged 53e1609db3cbb419277c5e04efd99df396428c3d into 5960b08e05ca243e863d7e95313d893a69f27536 )

**Pull request closed without merging**